### PR TITLE
add node.deleted to indicate if a node is deleted

### PR DESCRIPTION
**Pull request by mafintosh**
Backwards compat. Should make it possible to delete with a value which is useful for resolving conflicts

### CODE REVIEW

File: lib/iterator.js [hunk 1]
```diff
@@ -204,7 +204,8 @@ function byKey (a, b, reverse) {
 
 function allDeletes (nodes) {
   for (var i = 0; i < nodes.length; i++) {
-    if (nodes[i].value !== null) return false
+    if (nodes[i].value === null || nodes[i].deleted) continue
```
Comment (andrewosh) [inline lib/iterator.js]: So we're saying that `null` is still going to represent a deletion (as in, even with this addition one wouldn't be able to have meaningful `null` values)?

Comment (andrewosh) [reply lib/iterator.js]: Definitely necessary for backwards-compat, but is that a desirable limitation once this change is ancient history?

Comment (mafintosh) [reply lib/iterator.js]: No I'd like to get rid of that of check later on. Should we just drop support for `null === deleted` in 3.0.0 stable? I'm okay doing that

Comment (andrewosh) [reply lib/iterator.js]: Yeah especially since 3.0.0 is considered an "LTS" release right?

Comment (mafintosh) [reply lib/iterator.js]: yes, i'll update that

**Comment by mafintosh** [comment]
@andrewosh updated. wdyt?

File: lib/put.js [hunk 2]
```diff
@@ -44,6 +44,9 @@ PutRequest.prototype._finalize = function () {
     clock: this._clock
   }
 
+  // TODO: allow to set this flag explicitly
+  if (node.value === null) node.deleted = true
```
Comment (andrewosh) [inline lib/put.js]: Still want this one in there?

Comment (mafintosh) [reply lib/put.js]: Ah yea for now I need to keep it as that's how deletes are implemented internally (simply as `db.put(key, null)`). We can rewrite that at some point though

Comment (mafintosh) [reply lib/put.js]: Yea they'll turn into a delete (but we'll fix that asap)

**Pull request merged**